common/include/pcl/point_cloud.h

```diff
@@ -618,6 +618,14 @@ namespace pcl
       makeShared () const { return Ptr (new PointCloud<PointT> (*this)); }
 
     protected:
+      /**
+       * \brief Protected, non-virtual destructor
+       * \details This prevents deletion of a sub-class via pointer to the base-class. It is
+       * needed to throw a compile-time error when such undefined behavior is detected.
+       * The bitter alternative is to use a public virtual destructor
```

Comment:
Under which compiler exactly? I just tried the following snippet on [godbolt](https://godbolt.org/z/a8tF3q) and no compiler is throwing errors
```cpp
struct Base {
    ~Base() = default;
};


struct Derived : Base {
    int a;
};


int main()
{
    Base* d = static_cast<Base*>(new Derived);
    delete d;
    return 0;
}
```

Reply:
You missed to make the [dtor protected](https://godbolt.org/z/tfcNvf)